rasterio/_shim.pxd

```diff
@@ -1,5 +1,8 @@
 include "gdal.pxi"
 
+cdef GDALDatasetH open_dataset(object filename, int mode,
```

Comment:
Always adding the `GDAL_OF_RASTER` flag inside `open_dataset()` might be a bit safer rather than depending on the caller to _always_ set it.